src/sass_context.cpp

```diff
@@ -375,6 +375,11 @@ extern "C" {
     return options;
   }
 
+  const char* ADDCALL sass_copy_c_string (const char* input_string)
```

Comment:
The string passed to sass_copy_c_string will be deallocated by the called. The cloned string by strdup is returned to caller as a string, which will further passed as char* to sass_make_data_context. On deleting the context, libsass free this string. So everything is freed in the end.